src/main/java/blusunrize/immersiveengineering/common/blocks/metal/DieselGeneratorBlockEntity.java

```diff
@@ -158,32 +161,33 @@ public void tickServer()
 
 		if(!isRSDisabled()&&!tanks[0].getFluid().isEmpty())
 		{
-			GeneratorFuel recipe = recipeGetter.apply(level, tanks[0].getFluid().getFluid());
-			if(recipe!=null)
+			int output = IEServerConfig.MACHINES.dieselGen_output.get();
+			List<IEnergyStorage> presentOutputs = outputs.stream()
+					.map(CapabilityReference::getNullable)
+					.filter(Objects::nonNull)
+					.collect(Collectors.toList());
+			if(!presentOutputs.isEmpty()&&EnergyHelper.distributeFlux(presentOutputs, output, false) < output)
 			{
-				int burnTime = recipe.getBurnTime();
-				int fluidConsumed = FluidAttributes.BUCKET_VOLUME/burnTime;
-				int output = IEServerConfig.MACHINES.dieselGen_output.get();
-				List<IEnergyStorage> presentOutputs = outputs.stream()
-						.map(CapabilityReference::getNullable)
-						.filter(Objects::nonNull)
-						.collect(Collectors.toList());
-				if(!presentOutputs.isEmpty()&&
-						tanks[0].getFluidAmount() >= fluidConsumed&&
-						// Sort receivers by lowest input
-						EnergyHelper.distributeFlux(presentOutputs, output, false) < output)
+				consumeTick--;
+			}
+			if (consumeTick == 0) //Consume 10*tamount every 10t to allow for 1/10th mB amounts
+			{
+				GeneratorFuel recipe = recipeGetter.apply(level, tanks[0].getFluid().getFluid());
+				if(recipe!=null)
 				{
-					if(!active)
-					{
-						active = true;
-						animation_fanFadeIn = 80;
+					int burnTime = recipe.getBurnTime();
+					int fluidConsumed = (10*FluidAttributes.BUCKET_VOLUME)/burnTime;
+					if(tanks[0].getFluidAmount() >= fluidConsumed) {
+						if(!active) {
+							active = true;
+							animation_fanFadeIn = 80;
+						}
+						tanks[0].drain(fluidConsumed, FluidAction.EXECUTE);
+						consumeTick = 10;
+					} else if(active) {
+						active = false;
+						animation_fanFadeOut = 80;
 					}
```

Comment:
Code style please.
Opening and closing brackets go in separate lines.

Reply:
Hot damn I thought I fixed all those >.<